lib/ui/compositing.dart

```diff
@@ -79,16 +79,18 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
   ///
   /// Rasterization outside the given rounded rectangle is discarded.
   ///
-  /// See [pop] for details about the operation stack.
-  void pushClipRRect(RRect rrect) => _pushClipRRect(rrect._value);
-  void _pushClipRRect(Float32List rrect) native 'SceneBuilder_pushClipRRect';
+  /// See [pop] for details about the operation stack, and [Clip] for different clip modes.
+  /// By default, the clip will be anti-aliased (clipMode = 1 = Clip::antiAlias).
+  void pushClipRRect(RRect rrect, [int clipMode = 1]) => _pushClipRRect(rrect._value, clipMode);
```

Comment:
Can clipMode be an enum defined in the engine Dart code instead of an int?

Reply:
yeah the public Dart API should definitely not be using an `int` for this.

Reply:
also, prefer named arguments please

Reply:
Oh, I thought that only painting.dart is public and compositing.dart is private. If it's also public, then I shall definitely move the `enum Clip` definition from flutter/flutter's `basic_types.dart` into flutter/engine's `painting.dart`.